### DISCUSSION TITLE
Next.js deployed on Vercel - occasional server issues with new users on MongoDB

### DISCUSSION THREAD

**Discussion by allenwu25** [vercel/vercel#5128] [Help]
We have a site running on Next.js deployed on Vercel. Currently we have the issue where new users on the site get occasional 502/500 errors when they perform various functionality such as login, signup or viewing items. However, once they refresh the site it works fine again. When I checked the logs on Vercel, the issue appears as follows:
![image](https://user-images.githubusercontent.com/33672049/91670572-d1b98800-eaec-11ea-8255-489b78ba687c.png)

In MongoDB, I have set the Network Access to allow all addresses.
![image](https://user-images.githubusercontent.com/33672049/91670570-cb2b1080-eaec-11ea-8fa8-ace0e2d66ec3.png)

In addition, below is our connectDB function, and an example of where it is called. What might be causing this issue and how could it be resolved? Thanks

![image](https://user-images.githubusercontent.com/33672049/91670565-bfd7e500-eaec-11ea-9575-093f31e454e2.png)
![image](https://user-images.githubusercontent.com/33672049/91670559-b6e71380-eaec-11ea-8b62-c2d32a96f8e4.png)

**Comment by maxfi** [comment]
@allenwu25 I'm having the same issue. Did you figure out a fix for this? 🙏

**Reply by apsquared** [reply]
I'm also having the same issue and losing my mind.  

Just curious are you using the app router or pages router?
